### PR TITLE
build: fix build on go 1.13

### LINKED ISSUE #6603: go-critic: invalid pseudo-version: does not match version-control timestamp
#### Version information:

```yaml
OS: Android 10 aarch64
Host: google Pixel 3
Kernel: 4.9.165-gfff483291c07-ab5725533
Uptime: secs
Packages: 103 (dpkg), 1 (pkg)
Shell: bash 5.0.9
CPU: Qualcomm SDM845 (8) @ 1.766GHz
Memory: 2091MiB / 3546MiB
```

#### Description:

Error when trying to build.
```
go: github.com/go-critic/go-critic@v0.0.0-20181204210945-ee9bf5809ead: invalid pseudo-version: does not match version-control timestamp (2019-02-10T22:04:43Z)
go version go1.13rc1 android/arm64
bin/check_go_version 1.12
go install  "-asmflags=all='-trimpath='" "-gcflags=all='-trimpath='" -ldflags="-X "github.com/ipfs/go-ipfs".CurrentCommit=4a87fea33" ./cmd/ipfs
go: github.com/go-critic/go-critic@v0.0.0-20181204210945-ee9bf5809ead: invalid pseudo-version: does not match version-control timestamp (2019-02-10T22:04:43Z)
make: *** [cmd/ipfs/Rules.mk:37: cmd/ipfs-install] Error 1
```

### PR DESCRIPTION
**Pull request by Stebalien**
fixes #6603

### CODE REVIEW

**Comment by NatoBoram** [comment]
```
$ make install
go: github.com/golangci/golangci-lint@v1.17.1 requires
        github.com/golangci/errcheck@v0.0.0-20181003203344-ef45e06d44b6: invalid pseudo-version: does not match version-control timestamp (2018-12-23T08:41:20Z)
go version go1.13rc1 android/arm64
bin/check_go_version 1.12
go install  "-asmflags=all='-trimpath='" "-gcflags=all='-trimpath='" -ldflags="-X "github.com/ipfs/go-ipfs".CurrentCommit=bd6469860" ./cmd/ipfs
go: github.com/golangci/golangci-lint@v1.17.1 requires
        github.com/golangci/errcheck@v0.0.0-20181003203344-ef45e06d44b6: invalid pseudo-version: does not match version-control timestamp (2018-12-23T08:41:20Z)
make: *** [cmd/ipfs/Rules.mk:37: cmd/ipfs-install] Error 1
```

```
$ git status
HEAD detached at bd6469860
nothing to commit, working tree clean
```

File: go.mod [hunk 1]
```diff
@@ -110,8 +109,12 @@ require (
 	go.uber.org/multierr v1.1.0 // indirect
 	go4.org v0.0.0-20190313082347-94abd6928b1d // indirect
 	golang.org/x/sys v0.0.0-20190626221950-04f50cda93cb
+	golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7 // indirect
+	google.golang.org/appengine v1.4.0 // indirect
 	gopkg.in/cheggaaa/pb.v1 v1.0.28
 	gotest.tools/gotestsum v0.3.4
 )
 
+replace github.com/go-critic/go-critic v0.0.0-20181204210945-1df300866540 => github.com/go-critic/go-critic v0.3.5-0.20190526074819-1df300866540
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply go.mod]: https://tip.golang.org/doc/go1.13#version-validation

[the remark this answers is not in the document]
Comment (aschmahmann) [reply go.mod]: I see this has now been removed, but instead we have a replace directive for

`replace github.com/golangci/golangci-lint => github.com/mhutchinson/golangci-lint v1.17.2-0.20190819125825-d18f2136e32b`

If we need a replace directive in order to get things to work, why not just use the fork instead? Given that there are people using go-ipfs as a library this is just going to be annoying for them.

**Comment by Stebalien** [comment]
@NatoBoram does it work now?

**Comment by NatoBoram** [comment]
```yaml
go-ipfs version: 0.5.0-dev-45d6d9dcc
Repo version: 7
System version: arm64/android
Golang version: go1.13rc1
```

It works!

**Comment by kpp** [comment]
plz make a hotfix. Our code does not compile.

**Comment by Stebalien** [comment]
For now, you can fix this using a replace directive:

```
replace github.com/golangci/golangci-lint => github.com/golangci/golangci-lint v1.18.0
replace github.com/go-critic/go-critic v0.0.0-20181204210945-ee9bf5809ead => github.com/go-critic/go-critic v0.3.5-0.20190526074819-1df300866540
```

But yeah, I'm going to cut a patch release to fix this.

**Pull request merged**